homeassistant/components/homematicip_cloud/lock.py

```diff
@@ -0,0 +1,111 @@
+"""Support for HomematicIP Cloud lock devices."""
+from __future__ import annotations
+
+from typing import Any
+
+from homematicip.aio.device import AsyncDoorLockDrive
+from homematicip.base.enums import LockState, MotorState
+
+from homeassistant.components.lock import LockEntity, LockEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN as HMIPC_DOMAIN, HomematicipGenericEntity
+from .hap import HomematicipHAP
+
+ATTR_AUTO_RELOCK_DELAY = "auto_relock_delay"
+ATTR_DOOR_HANDLE_TYPE = "door_handle_type"
+ATTR_DOOR_LOCK_DIRECTION = "door_lock_direction"
+ATTR_DOOR_LOCK_NEUTRAL_POSITION = "door_lock_neutral_position"
+ATTR_DOOR_LOCK_TURNS = "door_lock_turns"
+
+DEVICE_DLD_ATTRIBUTES = {
+    "autoRelockDelay": ATTR_AUTO_RELOCK_DELAY,
+    "doorHandleType": ATTR_DOOR_HANDLE_TYPE,
+    "doorLockDirection": ATTR_DOOR_LOCK_DIRECTION,
+    "doorLockNeutralPosition": ATTR_DOOR_LOCK_NEUTRAL_POSITION,
+    "doorLockTurns": ATTR_DOOR_LOCK_TURNS,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the HomematicIP locks from a config entry."""
+    hap = hass.data[HMIPC_DOMAIN][config_entry.unique_id]
+    entities: list[HomematicipGenericEntity] = []
+    for device in hap.home.devices:
+        if isinstance(device, AsyncDoorLockDrive):
+            entities.append(HomematicipDoorLockDrive(hap, device))
+
+    if entities:
+        async_add_entities(entities)
```

Comment:
You could simplify the adding of entities with a comprehension:

```python
async_add_entities(
    HomematicipDoorLockDrive(hap, device)
    for device in hap.home.devices
    if isinstance(device, AsyncDoorLockDrive)
)
```

Reply:
There will be another lock sensor. So maybe it gets to complicated to read?

Reply:
Can you please explain, what i have done there? 😆 

The sensor is called HmIP-DLS and is just a sensor on the state of the lock. It has lockStates, but no motor to change them. I am not yet sure in which device class / platform i gonna put that device. I think putting it in lock platform could be the wrong place. Maybe sensor platform.